File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/jetstream/ProductClient.kt
package io.micronaut.nats.docs.jetstream

import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.messaging.annotation.MessageBody
import io.micronaut.nats.annotation.Subject
import io.micronaut.nats.jetstream.annotation.JetStreamClient
import io.nats.client.PublishOptions
import io.nats.client.api.PublishAck
// end::imports[]

@Requires(property = "spec.name", value = "JetstreamSpec")
// tag::clazz[]
@JetStreamClient
interface ProductClient {

    fun send(@Subject subject: String, @MessageBody data: ByteArray, publishOptions: PublishOptions): PublishAck // <1>
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/consumer/queue/ProductClient.py
from abc import ABC, abstractmethod

from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.nats.annotation import NatsClient, Subject
# end::imports[]


@Requires(property="spec.name", value="QueueSpec")
# tag::clazz[]
@NatsClient
class ProductClient(ABC):

    @Subject(value="product")  # <1>
    @abstractmethod
    def send(self, data: bytes) -> None:
        ...
# end::clazz[]
